pkg/collect/host_filesystem_performance.go

```diff
@@ -90,3 +91,234 @@ func (f FSPerfResults) String() string {
+		P90:     time.Duration(s.LatNs.Percentile.P90),
+		P95:     time.Duration(s.LatNs.Percentile.P95),
+		P99:     time.Duration(s.LatNs.Percentile.P99),
+		P995:    time.Duration(s.LatNs.Percentile.P995),
+		P999:    time.Duration(s.LatNs.Percentile.P999),
+		P9995:   time.Duration(s.LatNs.Percentile.P9995),
+		P9999:   time.Duration(s.LatNs.Percentile.P9999),
+	}
+}
+
+type FioNS struct {
+	Min        int64         `json:"min,omitempty"`
+	Max        int64         `json:"max,omitempty"`
+	Mean       float32       `json:"mean,omitempty"`
+	StdDev     float32       `json:"stddev,omitempty"`
+	N          int64         `json:"N,omitempty"`
+	Percentile FioPercentile `json:"percentile,omitempty"`
+}
+
+type FioDepth struct {
+	FioDepth0    float32 `json:"0,omitempty"`
+	FioDepth1    float32 `json:"1,omitempty"`
+	FioDepth2    float32 `json:"2,omitempty"`
+	FioDepth4    float32 `json:"4,omitempty"`
+	FioDepth8    float32 `json:"8,omitempty"`
+	FioDepth16   float32 `json:"16,omitempty"`
+	FioDepth32   float32 `json:"32,omitempty"`
+	FioDepth64   float32 `json:"64,omitempty"`
+	FioDepthGE64 float32 `json:">=64,omitempty"`
+}
+
+type FioLatency struct {
+	FioLat2      float32 `json:"2,omitempty"`
+	FioLat4      float32 `json:"4,omitempty"`
+	FioLat10     float32 `json:"10,omitempty"`
+	FioLat20     float32 `json:"20,omitempty"`
+	FioLat50     float32 `json:"50,omitempty"`
+	FioLat100    float32 `json:"100,omitempty"`
+	FioLat250    float32 `json:"250,omitempty"`
+	FioLat500    float32 `json:"500,omitempty"`
+	FioLat750    float32 `json:"750,omitempty"`
+	FioLat1000   float32 `json:"1000,omitempty"`
+	FioLat2000   float32 `json:"2000,omitempty"`
+	FioLatGE2000 float32 `json:">=2000,omitempty"`
+}
+
+type FioDiskUtil struct {
+	Name        string  `json:"name,omitempty"`
+	ReadIos     int64   `json:"read_ios,omitempty"`
+	WriteIos    int64   `json:"write_ios,omitempty"`
+	ReadMerges  int64   `json:"read_merges,omitempty"`
+	WriteMerges int64   `json:"write_merges,omitempty"`
+	ReadTicks   int64   `json:"read_ticks,omitempty"`
+	WriteTicks  int64   `json:"write_ticks,omitempty"`
+	InQueue     int64   `json:"in_queue,omitempty"`
+	Util        float32 `json:"util,omitempty"`
+}
+
+type FioPercentile struct {
+	P1    int `json:"1.000000,omitempty"`
+	P5    int `json:"5.000000,omitempty"`
+	P10   int `json:"10.000000,omitempty"`
+	P20   int `json:"20.000000,omitempty"`
+	P30   int `json:"30.000000,omitempty"`
+	P40   int `json:"40.000000,omitempty"`
+	P50   int `json:"50.000000,omitempty"`
+	P60   int `json:"60.000000,omitempty"`
+	P70   int `json:"70.000000,omitempty"`
+	P80   int `json:"80.000000,omitempty"`
+	P90   int `json:"90.000000,omitempty"`
+	P95   int `json:"95.000000,omitempty"`
+	P99   int `json:"99.000000,omitempty"`
+	P995  int `json:"99.500000,omitempty"`
+	P999  int `json:"99.900000,omitempty"`
+	P9995 int `json:"99.950000,omitempty"`
+	P9999 int `json:"99.990000,omitempty"`
+}
+
+func (d FioDiskUtil) Print() string {
```

Comment:
Change this to `func (d FioDiskUtil) Print() string {`. Its Go's standard way of "stringifying" an object. Once you make that change, any string formatting will detect the new `String()` method i.e `fmt.Sprintf("%s\n", object)`

Reply:
@banjoh ty!